### PR TITLE
A new Adherent can now resume the subscription process after donating

### PR DESCRIPTION
**Pull request by rpg600**

### CODE REVIEW

File: src/AppBundle/Controller/DonationController.php [hunk 1]
```diff
@@ -116,11 +116,14 @@ public function resultAction(Request $request, Donation $donation)
 
         $retryUrl = $this->generateUrl('donation_informations', $parameters);
 
+        $membershipUtils = $this->get('app.membership_utils');
```
Comment (tgalopin) [inline src/AppBundle/Controller/DonationController.php]: Is a variable really needed here? `$this->get('app.membership_utils')->isInSubscriptionProcess()` seems fine to me.

File: src/AppBundle/Membership/MembershipUtils.php [hunk 2]
```diff
@@ -47,11 +47,21 @@ public function clearRegisteringDonation()
      *
      * @return int|null
```
Comment (tgalopin) [inline src/AppBundle/Membership/MembershipUtils.php]: Can you remove this now useless phpdoc?

File: src/AppBundle/Membership/MembershipUtils.php [hunk 3]
```diff
@@ -47,11 +47,21 @@ public function clearRegisteringDonation()
...
-    public function getNewAdherentId()
+    public function getNewAdherentId(): ?int
     {
         return $this->session->get(self::NEW_ADHERENT_ID);
     }
 
+    /**
+     * If true, the current user has started the subscription process but hasn't finished yet.
+     *
+     * @return bool
```
Comment (tgalopin) [inline src/AppBundle/Membership/MembershipUtils.php]: Return type useless

File: app/Resources/views/donation/result.html.twig [hunk 4]
```diff
@@ -53,6 +53,13 @@
                     <i class="fa fa-twitter"></i> <span>Je tweete</span>
                 </li>
             </ul>
+
+            {% if isInSubscriptionProcess %}
```
Comment (hhamon) [inline app/Resources/views/donation/result.html.twig]: Twig variables must be in snake_case.

File: src/AppBundle/Controller/DonationController.php [hunk 5]
```diff
@@ -121,6 +121,7 @@ public function resultAction(Request $request, Donation $donation)
             'error_code' => $request->query->get('code'),
             'donation' => $donation,
             'retry_url' => $retryUrl,
+            'isInSubscriptionProcess' => $this->get('app.membership_utils')->isInSubscriptionProcess(),
```
Comment (hhamon) [inline src/AppBundle/Controller/DonationController.php]: snake_case

**Pull request merged**